### PR TITLE
baremetal: add documentation for configuration validations

### PR DESCRIPTION
**Pull request by andfasano**
This PR enhances the documentation for the baremetal configuration by describing the validations added in #3392, #3232 and #3358

### CODE REVIEW

File: docs/user/metal/install_ipi.md [hunk 1]
```diff
@@ -197,7 +197,7 @@ platform:
           username: admin
           password: password
         bootMACAddress: 00:11:07:4e:f6:71
-        hardwareProfile: default
+        hardwareProfile: default
```
Comment (stbenjam) [inline docs/user/metal/install_ipi.md:200]: Some whitespace snuck in, can you remove it? Otherwise LGTM, thanks!

Comment (andfasano) [reply docs/user/metal/install_ipi.md]: Removed

**Pull request merged**